File: OrdenarArquivos/src/main/java/com/fenix/ordenararquivos/model/Caminhos.java
package com.fenix.ordenararquivos.model;

public class Caminhos {

	private Long id;
	private Manga manga;
	private String capitulo;
	private Integer numero;
	private String numeroPagina;
	private String nomePasta;
	
	
	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Manga getManga() {
		return manga;
	}

	public void setManga(Manga manga) {
		this.manga = manga;
	}

	public String getCapitulo() {
		return capitulo;
	}

	public void setCapitulo(String capitulo) {
		this.capitulo = capitulo;
	}

	public Integer getNumero() {
		return numero;
	}
	
	public String getNumeroPagina() {
		return numeroPagina;
	}

	public void setNumero(Integer numero) {
		this.numero = numero;
		this.numeroPagina = numero.toString();
	}
	
	public void setNumero(String numeroPagina) {
		this.numero = numeroPagina == null || numeroPagina.isEmpty() ? 0 : Integer.valueOf(numeroPagina);
		this.numeroPagina = numeroPagina;
	}

	public String getNomePasta() {
		return nomePasta;
	}

	public void setNomePasta(String nomePasta) {
		this.nomePasta = nomePasta;
	}

	public Caminhos() {

	}

	public Caminhos(String capitulo, Integer numero, String nomePasta) {
		this.capitulo = capitulo;
		this.numero = numero;
		this.numeroPagina = numero.toString();
		this.nomePasta = nomePasta;
	}
	
	public Caminhos(String capitulo, String numero, String nomePasta) {
		this.capitulo = capitulo;
		this.numero = numero == null || numero.isEmpty() ? 0 : Integer.valueOf(numero);
		this.numeroPagina = numero;
		this.nomePasta = nomePasta;
	}

	public Caminhos(Long id, Manga manga, String capitulo, Integer numero, String nomePasta) {
		this.id = id;
		this.manga = manga;
		this.capitulo = capitulo;
		this.numero = numero;
		this.numeroPagina = numero.toString();
		this.nomePasta = nomePasta;
	}

	@Override
	public String toString() {
		return "Caminhos [capitulo=" + capitulo + ", numero=" + numero + ", nomePasta=" + nomePasta + "]";
	}

}

File: OrdenarArquivos/src/main/kotlin/com/fenix/ordenararquivos/model/entities/Caminhos.kt
package com.fenix.ordenararquivos.model.entities

import com.google.firebase.database.Exclude

data class Caminhos(
    var id: Long = 0,
    @Exclude @set:Exclude @get:Exclude var manga: Manga? = null,
    var capitulo: String = "",
    private var _numero: Int = 0,
    private var _numeroPagina: String = _numero.toString(),
    var nomePasta: String = ""
) {

    var numero: Int = _numero
        set(value) {
            numeroPagina = value.toString()
            field = value
        }

    fun addNumero(numero: String) {
        this.numero = if (numero.isEmpty()) 0 else Integer.valueOf(numero)
    }

    var numeroPagina: String = _numeroPagina
        private set

    constructor(capitulo: String, numero: String, nomePasta: String) : this(capitulo = capitulo, nomePasta = nomePasta, _numero = if (numero.isEmpty()) 0 else Integer.valueOf(numero)) { }

    constructor(id: Long, manga: Manga, capitulo: String, pagina: Int, pasta: String) : this(id, manga, capitulo, pagina, pagina.toString(), pasta) {}

    override fun toString(): String {
        return "Caminhos [capitulo=$capitulo, numero=$numero, nomePasta=$nomePasta]"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Caminhos

        if (manga != other.manga) return false
        if (capitulo != other.capitulo) return false
        if (nomePasta != other.nomePasta) return false

        return true
    }

    override fun hashCode(): Int {
        var result = manga?.hashCode() ?: 0
        result = 31 * result + capitulo.hashCode()
        result = 31 * result + nomePasta.hashCode()
        return result
    }

}